tests/data/serde/README.md

```diff
@@ -0,0 +1,12 @@
+Serialization input/output
+==========================
+
+Files here contain hex strings and binary data representing types used for
+regression testing.
+
+- *_hex: consensus encoded types represented as hex strings
```

Comment:
Why do we need `_hex` files? Shouldn't we test hex decoding separately?

Reply:
TBH I'd rather drop the binary vectors than the hex ones.

Agreed that they're pretty-much redundant and we shouldn't duplicate everything.

Reply:
hmm, I can't really remember why I put some in hex (I did the bulk of this work in November last year) but my guess was that I was lazy and just used hex strings when they appeared elsewhere in the codebase (the binary stuff I generated for each type manually). I do remember going through at the end and moving the hex stuff to files instead of inline. My reasoning was that the data is not meaningful so better to conceal it in a file and keep the test source code clean. I can't remember why I used binary instead of hex for the manually generated types, perhaps because we have the two blocks there in `tests/data` that are in binary - if we move to hex we should be uniform. We exclude the `tests/` directory in `Cargo.toml` so data size is not a consideration. I do not know what other trade offs there are between the two?

Reply:
I think it's fine -- usually they'll just be "parsing" the hex to a `Vec<u8>` then parsing from there as though it were binary data. And hex->byte vector is write-once code.

Reply:
Yesterday I forgot that the hex files hold hex that is consensus excodings, we are consensus deserialize them, serde serializing and then checking the serder serializaiton (using `bincode` crate) against the data in the _bincode files. Are you guys happy with this method?

An alternate method would be have a single hex file and serde deserialize it, serde serialize it, an compare - I do not know if one method is better than the other but my gut feel is that the current method is better.

I want to confirm before changing everything please because this PR is as boring to work on as it is to review :)

Reply:
I think it makes sense to do cross-serialization tests so that if there was the same bug in serde serailization and deserialization it wouldn't be missed. So no change in method. If you both prefer hex to binary it's OK.

Reply:
So do you want me to remove the use of `bincode` crate for doing serde serialization @apoelstra and use the `hex` crate instead?

Reply:
We should never need the `hex` crate.

We should probably add a `deserialize_hex` method analogous to `serialize_hex` which uses the `bitcoin_hashes::hex` module.

Reply:
I don't get it, we are explicitly trying to test the serde implementations, we have to use a crate serializes using serde to do so, right?

Reply:
I was not aware that `hex` uses serde. What serialization format does it use?

Reply:
Looks like it just serializes as ascii chars: https://docs.rs/hex/latest/hex/serde/fn.serialize.html

Reply:
It's simple enough for us to NIH it.